internal/server/middleware.go

```diff
@@ -80,27 +80,13 @@ func authenticatedMiddleware(srv *Server) gin.HandlerFunc {
 				return
 			}
 
-			// Before changing anything here, make sure you read the org-selection workflow diagram.
-			// https://github.com/infrahq/infra/blob/main/docs/dev/organization-request-flow.md
```

Comment:
I moved this link into `unauthenticatedMiddleware` where we check if org is unset, now that #2968 has updated the diagram for that case.